mantidproject/mantid
# Create circular beam profile

**Description of work.**
Create a Circular Beam profile and provide support in the SetBeam, MonteCarlo and AddAbsorptionWeightedPathLengths algorithms.

**Report to:** Danny Hindson, Chris Ridley


**To test:**

[29001_test.zip](https://github.com/mantidproject/mantid/files/5109407/29001_test.zip)

The attached script runs a MonteCarloAbsorption using both a Slit (0.5x0.5) and Circular (radius=0.5) beam profile, with two separate output workspaces.

1. Download and run the test script
2. Plot and overplot the two output workspaces
3. Check that there are small differences in the results but that the plot shape remains mostly similar

Fixes #29001.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
